planner/core/rule_column_pruning.go

```diff
@@ -221,17 +223,13 @@ func (ds *DataSource) PruneColumns(parentUsedCols []*expression.Column) error {
 	exprCols := expression.ExtractColumnsFromExpressions(nil, ds.allConds, nil)
 	exprUsed := expression.GetUsedList(exprCols, ds.schema)
 
-	var (
-		handleCol     *expression.Column
-		handleColInfo *model.ColumnInfo
-	)
-	if ds.handleCol != nil {
-		handleCol = ds.handleCol
-		handleColInfo = ds.Columns[ds.schema.ColumnIndex(handleCol)]
-	}
 	originSchemaColumns := ds.schema.Columns
 	originColumns := ds.Columns
 	for i := len(used) - 1; i >= 0; i-- {
+		if ds.tableInfo.IsCommonHandle && mysql.HasPriKeyFlag(ds.schema.Columns[i].RetType.Flag) {
+			// Do not prune common handle column.
```

Comment:
Why we don't prune common columns in schema?

Reply:
For simplicity.